cluster/src/dunit/scala/io/snappydata/metrics/SnappyMetricsSystemDUnitTest.scala

```diff
@@ -119,13 +111,11 @@ class SnappyMetricsSystemDUnitTest(s: String)
     found
   }
 
-  def getConnection(): Connection = {
-    val driver = "io.snappydata.jdbc.ClientDriver"
-    Utils.classForName(driver).newInstance
-    DriverManager.getConnection(s"jdbc:snappydata:thrift://localhost[$netPort]")
-  }
+  def getConnection(netPort: Int): Connection =
+    DriverManager.getConnection(s"${Constant.DEFAULT_THIN_CLIENT_URL}localhost:$netPort")
 
   def testMetricsMonitoring(): Unit = {
+    Thread.sleep(10000)
```

Comment:
Why is this sleep for and can we instead use the `WaitCriterion`?

Reply:
Sleep is required to take effects in metrics values as the metrics values gets updated by 5 sec